src/quarto/game/game.go

```diff
@@ -0,0 +1,82 @@
+package game
+
+const GridSize = 4
+
+// State define data for a game state
+type State struct {
+	Grid  [GridSize][GridSize]int
+	Piece int
+}
+
+// DoAMove return the next move for given grid
+func DoAMove(state State) State {
+
+	state = PlacePieceOnGrid(state)
```

Comment:
Please do not change the state passed in parameter but create a new instance of State.
You functions must be immutable.